features/validator.feature

```diff
@@ -0,0 +1,138 @@
+    Given a valid file with filename <file_name>
+    And format field set to <file_format>
+    When metadata is validated
+    Then File is <validation_state> after validation
+
+    Examples:
+      |           file_name |       file_format | validation_state |
+      |       file.fastq.gz |             fq.gz |          INVALID |
+      |          file.fq.gz |          fastq.gz |          INVALID |
+      |          file.fastq |             fq.gz |          INVALID |
+      |   file.fastq.tar.gz |          fastq.gz |          INVALID |
+
+  Scenario Outline: 2.a.2 invalid fastq file - filename extension doesn't match the file format - both point to fastq validator
+    Given an invalid file with filename <file_name>
+    And format field set to <file_format>
+    When metadata is validated
+    Then File is <validation_state> after validation
+
+    Examples:
+      |           file_name |   file_format | validation_state |
+      |       file.fastq.gz |         fq.gz |          INVALID |
+      |          file.fq.gz |      fastq.gz |          INVALID |
+      |          file.fastq |         fq.gz |          INVALID |
+      |   file.fastq.tar.gz |      fastq.gz |          INVALID |
+
+  Scenario Outline: 2.b.1 filename extension doesn't match the file format -- filename extension pointing to fastq validator - file format not blank
+    Given a valid file with filename <file_name>
+    And format field set to <file_format>
+    When metadata is validated
+    Then File is <validation_state> after validation
+
+    Examples:
+      |           file_name | file_format | validation_state |
+      |       file.fastq.gz |         bam |          INVALID |
+      |          file.fq.gz |         txt |          INVALID |
+      |          file.fastq |         pdf |          INVALID |
+      |   file.fastq.tar.gz |         doc |          INVALID |
+
+  Scenario Outline: 2.b.2 filename extension doesn't match the file format -- filename extension pointing to fastq validator - file format is blank
+    Given a valid file with filename <file_name>
+    And format field set to <file_format>
+    When metadata is validated
+    Then File is <validation_state> after validation
+
+    Examples:
+      |           file_name |       file_format | validation_state |
+      |       file.fastq.gz |                   |            VALID |
+      |          file.fq.gz |                   |            VALID |
+      |          file.fastq |                   |            VALID |
+      |   file.fastq.tar.gz |                   |            VALID |
+      |             file.fq |                   |            VALID |
+
+  Scenario Outline: 2.b.3 filename extension doesn't match the file format -- filename extension pointing to fastq validator - file format is blank - invalid file
+    Given an invalid file with filename <file_name>
+    And format field set to <file_format>
```

Comment:
instead of having an empty table column, why no have a step called "and en empty field format field"

Reply:
makes sense! thanks!